### ISSUE TITLE
Find occurrences doesn't work when string contains non-alpha numeric chars

### ISSUE THREAD

**Issue by dflock** [lapce/lapce#1692]
## Lapce Version

0.2.3

## System information

Windows 10

## Describe the bug

If the selected string contains non-alpha numeric chars, find occurrences does nothing. I would expect find occurrences to work with any string.

## Additional information

For example, given this text:

```
the <fox> the <fox>
the (fox) the (fox)
the fox. the fox.
the fox- the fox-

{}()<>!@#$%^&*+
{}()<>!@#$%^&*+
```

If you select the first `the`, then press ctrl+d, lapce will select the second `the`. This also works if you select the first `fox` (i.e. just the letters, not the whole `<fox>`). However, if you select the first `<fox>` (including the gt/lt symbols), pressing ctrl+d does nothing. This holds for `(fox)` and `fox.`

Find occurrences also fails if you select just a `(`, or any of the chars on the last two lines, for example.

Interestingly, underscores work.

**Comment by panekj** [comment]
search is regex-based so unescaped characters might not work

**Comment by mustafaquraish** [comment]
> search is regex-based so unescaped characters might not work

I would not expect regex-based search when using `ctrl+d`. Is there a way to escape automatically escape the characters?

**Comment by dflock** [comment]
Yeah, this was surprising behavior to me, which was why I filled a bug
report. I would expect find occurences/ctrl+d to be just simple string
matching.

**Comment by beckend** [comment]
@dflock just tested this with this setting disabled(unchecked):
Editor Settings: `Multicursor Whole Words`
Then it works for your usecase.

**Comment by dflock** [comment]
> @dflock just tested this with this setting disabled(unchecked): Editor Settings: `Multicursor Whole Words` Then it works for your usecase.

Can confirm, this makes it behave as I wold expect - and as other editors find occurences work.

This should definitely be the way that find occurences works by default.

**Comment by beckend** [comment]
I concur.
